_posts/2020-08-06-changelog-128.markdown

```diff
@@ -0,0 +1,171 @@
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5808 -->
+* Better type inference for `associated_type_bounds` rustc nightly feature
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5733 -->
+* Infer type for types implementing the `Try` trait when using question mark operator (by [@Kobzol])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5643 -->
+* Add `Substitute associated type` intention (by [@Kobzol])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5798 -->
+* Add `Main function not found in crate` ([E0601](https://doc.rust-lang.org/error-index.html#E0601)) inspection for build scripts (by [@Stzx])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5747 -->
+* Fold external (non-project related) parts of stack traces (by [@Kobzol])
+
+## Fixes
+
+* Type inference fixes:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5841 -->
+    * Infer type parameters from associated type bindings
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5837 -->
+    * Infer types of complex `?` expressions
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5824 -->
+    * Ignore currently unsupported negative `impl` blocks during type inference
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5812 -->
+    * Fix `self` substitution in bounds during type inference
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5809 -->
+    * Fix nested projections normalization
+    
+    
+* Macro expansion fixes:    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5862 -->
+    * Fix `ArrayIndexOutOfBoundsException` during macro call arguments completion
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5810 -->
+    * Replace `IntellijRustDollarCrate` with a crate name in `Show macro expansion` action
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5796 -->
+    * Resolve `$crate` via hygiene
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5855 -->
+    * Don't skip doc comments provided via macro definition
+    
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/5777 -->
+    * Fix groups substitution in macro expansion
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5857 -->
+* Find and replace occurrences located outside a function in `Introduce constant/variable` refactoring (by [@Kobzol])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5856 -->
+* Properly check spelling in lifetimes. Now `'static` is not reported as a typo
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5853 -->
+* Fix name resolution of private reexport with `crate::` path
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5849 -->
+* Include `RUSTC` in environment when executing Cargo (by [@tamird])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/5842 -->
+* Don't show parameter hints for single-parameter closures
```

Comment:
Let's add note about "smart" hint option to return the corresponding hints back if needed